app/uk/gov/hmrc/exports/services/mapping/declaration/BorderTransportMeansBuilder.scala

```diff
@@ -67,11 +67,13 @@ class BorderTransportMeansBuilder @Inject()(countriesService: CountriesService)
   }
 
   private def appendDepartureTransport(data: Transport, transportMeans: Declaration.BorderTransportMeans): Unit =
-    data.borderModeOfTransportCode.foreach { modeOfTransportCode =>
-      if (modeOfTransportCode.isDefined) {
-        val modeCode = new BorderTransportMeansModeCodeType()
-        modeCode.setValue(modeOfTransportCode.value)
-        transportMeans.setModeCode(modeCode)
-      }
+    for {
+      transportLeavingTheBorder <- data.borderModeOfTransportCode
+      modeOfTransportCode <- transportLeavingTheBorder.code
+    } if (modeOfTransportCode.isDefined) {
```

Comment:
you can move if to the for comprehension, this probably will be more readalbe. Combination for and if for me in this way is difficult to read:

```
for {
  transportLeavingTheBorder <- data.borderModeOfTransportCode
  modeOfTransportCode <- transportLeavingTheBorder.code
  if(modeOfTransportCode.isDefined)
} {
  // Rest of the code what should be done
}
```